rfcs/0016-permissionless-assets-for-ibc-and-cw.md

```diff
@@ -0,0 +1,77 @@
+
+### Ratio Manipulation
+
+Ratio will be set to `None` for all permissionless assets. It will then be set to some value via governance if asset is decided to be good for BYOG and paying for XCM transfers. Until then, 2 assets will be required to be sent via XCM: permissionless asset + asset to pay for transaction.
+
+### ED Manipulation
+
+Existential Deposit(ED) is a safeguard against bloating storage of the chain. If a user transfers a nonvaluable asset and sets its ED to 0 or 1, this user will be able to affect performance of the parachain by transferring this asset to many accounts. Thus, to avoid this potential attack vector, we will use Asset Hub (formerly Statemine) approach of non-sufficient assets: a user to be able to recieve permissionless asset needs to hold ED in PICA on Picasso and LAYR on Composable parachains. This approach will limit the growth of the storage. 
+
+The chain needs to make sure that permissionless assets can be transferred to accounts that dont have ED in parachain's native token. It is also required to fail transfers that will result in drop of native token below ED if this account has nonsufficient tokens. [Reference](https://substrate.stackexchange.com/questions/2447/influence-of-existential-deposits-on-account-assets)
+
+#### Non-sufficient assets implementation concerns
+
+#### Recieving Account doesnt hold pica during bridging
+**Potential solution**: relayer will send PICA to the reciever in case reciever doesnt have it. In case bridged asset is FlatFee asset, additional fee equal to ED in PICA will be charged from the transferred amount. Otherwise, percentage fee will be applied(To be configured)
+
+#### orml tokens pallet EDs
+Currently assets are being stored in orml tokens pallet which require Existential Deposits for stored assets. 
+
+**Potential Solution**: for non sufficient assets still provide non-zero Existential Deposit parameter which is similar to minimum balance in Asset Hub implementation. Then the Exitstential logic for orml tokens pallet will stay the same because account wont be able to hold a non sufficient asset without ED in parachain's native token. This will require a new field in AssetsRegistry: is_sufficient. All permissionless assets will have this field to be false by default. Assets created via permissioned extrinsic will allow to choose if asset is sufficient or not. Asset's sufficiency can be changed by governance.
+
+
+## Updated extrinsics
+
+Extrinsic to create asset via Root  `assets_registry.register_asset(name, symbol, decimals, location, ED, Ratio, is_sufficient)`.
+
+Extrinsic to create asset for bridging is `permissionless_assets.create_bridge_asset(name, symbol, decimals, location, ED)`.
+
+Extrinsic to create asset for CosmWasm is `permissionless_assets.create_cw_asset(name, symbol, decimals, ED)`.
```

Comment:
how it can set ed if ed is pica?  does tokenfactory allows to set ed and requires all parameters or some optional? we should be compatible

Reply:
Nonsufficent assets still have minimal balance on AssetHub (Statemine) which is basically ED. So ED of pica is to prevent spamming accounts , ED for asset will be for orml_tokens pallet. If account has enough of PICA ED, but nonsufficient asset amount drops below minimum balance(ED), then account stays alive, but orml tokens pallets clears the nonsufficent asset state for the account. The behavior is the same, so we dont need to get rid of ED(which is the same thing as minimum balance) for nonsufficient assets but add additional ED in PICA